bin/modules/calculate_properties.py

```diff
@@ -345,3 +349,590 @@ def getStructuresFromOptimization(outputfolder, FrameworkName) -> list:
 
     return structure_list
+
+
+def get_spg_class(spgnum) -> str:
+    '''
+    Get the space group class from the space group number.
+    Parameters
+    ----------
+    spgnum : int
+        Space group number
+
+    Returns
+    -------
+    spgclass : str
+        Space group class
+    '''
+    # Triclinic or monoclinic or orthorhombic
+    spgclass = ""
+    if (spgnum < 3):
+        spgclass = "triclinic"
+    elif (spgnum > 3 and spgnum < 16):
+        spgclass = "monoclinic"
+    elif (spgnum > 15 and spgnum < 75):
+        spgclass = "orthorhombic"
+    elif (spgnum > 74 and spgnum < 143):
+        spgclass = "tetragonal"
+    elif (spgnum > 142 and spgnum < 168):
+        spgclass = "trigonal"
+    elif (spgnum > 167 and spgnum < 195):
+        spgclass = "hexagonal"
+    elif (spgnum > 194):
+        spgclass = "cubic"
+
+    return spgclass
+
+
+def get_reciprocal_vectors(CellMatrix) -> tuple[float, float, float]:
+    '''
+    Get the reciprocal vectors of a cell given in cell parameters of cell vectors
+    ----------
+    CellMatrix : array
+        (3,1) array for cell vectors
+    Returns
+    -------
+    b1 : array
+        (3,1) array containing b_1 vector in the reciprocal space
+    b2 : array
+        (3,1) array containing b_2 vector in the reciprocal space
+    b3 : array
+        (3,1) array containing b_3 vector in the reciprocal space
+    '''
+
+    v1, v2, v3 = CellMatrix
+
+    vol = np.dot(v1, np.cross(v2, v3))
+
+    b1 = 2 * np.pi * np.cross(v2, v3) / vol
+    b2 = 2 * np.pi * np.cross(v3, v1) / vol
+    b3 = 2 * np.pi * np.cross(v1, v2) / vol
+
+    return b1, b2, b3
+
+
+def get_kgrid(cell, dist=0.3) -> tuple[float, float, float]:
+    '''Get the k-points grid in the reciprocal space with a given distance for a
+    cell given in cell parameters of cell vectors.
+    ----------
+    cell : array
+        (3,1) array for cell vectors or (6,1) array for cell parameters
+    distance : float
+        distance between the points in the reciprocal space
+    Returns
+    -------
+    kx : int
+        Number of points in the x direction on reciprocal space
+    ky : int
+        Number of points in the y direction on reciprocal space
+    kz : int
+        Number of points in the z direction on reciprocal space
+    '''
+
+    b1, b2, b3 = get_reciprocal_vectors(cell)
+
+    b = np.array([np.linalg.norm(b1),
+                  np.linalg.norm(b2),
+                  np.linalg.norm(b3)])
+
+    kx = np.ceil(b[0]/dist).astype(int)
+    ky = np.ceil(b[1]/dist).astype(int)
+    kz = np.ceil(b[2]/dist).astype(int)
+
+    return kx, ky, kz
+
+
+def create_input_file(FrameworkName: str,
+                      output_folder: str,
+                      **kwargs):
+    """
+    Create the input file for CP2K
+
+    Parameters
+    ----------
+    FrameworkName : str
+        Name of the framework
+    output_folder : str
+        Path to the output folder
+    """
+
+    CalcDict = {
```

Comment:
Why is it necessary to initialise `CalcDict` with default values? Is it because not all parameters are reflected in the `--argument` list?

If so, wouldn't it be easier to add these extra parameters directly to the `create_input_file` call just to avoid repeating the same dictionary twice? 

For example, `MaxIterations` is not a user-defined parameter in `bin/generate_shift.py` and always takes the default value of 100. Instead of having it here in `bin/modules/calculate_properties.py`, one could bring it to the surface (closer to the user) by simply adding `MaxIterations=100` in the list of arguments to `create_input_file()` in `bin/generate_shift.py`. That way we avoid having to always ensure that the default value of an argument is the same in `bin/generate_shift.py` and `bin/modules/calculate_properties.py`.